presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlock.java

```diff
@@ -193,4 +195,20 @@ public Block getLoadedBlock()
                 offsets,
                 loadedValuesBlock);
     }
+
+    @Override
+    public Block getBlockUnchecked(int position)
+    {
+        int startValueOffset = getOffsets()[position];
+        int endValueOffset = getOffsets()[position + 1];
+        return getRawElementBlock().getRegion(startValueOffset, endValueOffset - startValueOffset);
+    }
+
+    @Override
+    public boolean isNullUnchecked(int position)
```

Comment:
I'd like to rename `position` argument here to avoid confusion with the method above. I think the two values are different, one is logical position in the block, the other one is offsetBase + logical position.